### PR TITLE
Elemtypes map

### PR DESCRIPTION
**Pull request by joergbuchwald**
Substitute getNumberOfElementTypes() array by std::map to avoid multiple different element type mappings.

I'm not able to check the output of the DataExplorer right now as the binary makes some throws some OpenGL/QT errors on my machine. But at least it compiles.

### CODE REVIEW

File: Tests/MeshLib/TestAddLayerToMesh.cpp [hunk 1]
```diff
@@ -70,9 +70,9 @@ TEST(MeshLib, AddTopLayerToLineMesh)
     ASSERT_EQ(2*mesh->getNumberOfNodes(), result->getNumberOfNodes());
     ASSERT_EQ(2*mesh->getNumberOfElements(), result->getNumberOfElements());
 
-    std::array<unsigned, 7> const n_elems (MeshLib::MeshInformation::getNumberOfElementTypes(*result));
-    ASSERT_EQ(5, n_elems[0]); // tests if 5 lines are present
-    ASSERT_EQ(5, n_elems[2]); // tests if 5 quads are present
+    auto n_elems = MeshLib::MeshInformation::getNumberOfElementTypes(*result);
```
Comment (endJunction) [inline Tests/MeshLib/TestAddLayerToMesh.cpp:73]: keep const

File: MeshLib/MeshInformation.cpp [hunk 2]
```diff
@@ -25,58 +24,28 @@ GeoLib::AABB MeshInformation::getBoundingBox(const MeshLib::Mesh& mesh)
...
 void MeshInformation::writeAllNumbersOfElementTypes(const MeshLib::Mesh& mesh)
 {
-    auto const& nr_ele_types =
-        MeshLib::MeshInformation::getNumberOfElementTypes(mesh);
-    const std::array<std::string, 7> element_names{
-        "lines",     "triangles", "quads", "tetrahedra",
-        "hexahedra", "pyramids",  "prisms"};
-    for (unsigned int element_type = 0; element_type < nr_ele_types.size();
-         element_type++)
+    auto nr_ele_types = MeshLib::MeshInformation::getNumberOfElementTypes(mesh);
```
Comment (endJunction) [inline MeshLib/MeshInformation.cpp:43]: const&

File: MeshLib/MeshInformation.cpp [hunk 3]
```diff
@@ -25,58 +24,28 @@ GeoLib::AABB MeshInformation::getBoundingBox(const MeshLib::Mesh& mesh)
...
+
+    INFO("Number of elements in the mesh:")
+    for (std::map<MeshElemType, unsigned>::iterator it=nr_ele_types.begin(); it!=nr_ele_types.end(); ++it)
```
Comment (endJunction) [inline MeshLib/MeshInformation.cpp:46]: formatting

Comment (endJunction) [reply MeshLib/MeshInformation.cpp:46]: Better yet to use the for-range-loop.

File: MeshLib/MeshInformation.cpp [hunk 4]
```diff
@@ -25,58 +24,28 @@ GeoLib::AABB MeshInformation::getBoundingBox(const MeshLib::Mesh& mesh)
...
     {
-        INFO("\t%d %s ", nr_ele_types[element_type],
-             element_names[element_type].c_str());
+        INFO("\t%ss: %d ", MeshLib::MeshElemType2String(static_cast<MeshLib::MeshElemType>(it->first)).c_str(), it->second);
```
Comment (endJunction) [inline MeshLib/MeshInformation.cpp:48]: Is the double s correct? `%ss`

Comment (joergbuchwald) [reply MeshLib/MeshInformation.cpp:48]: That's correct. Because the casting generates singular typenames, but I want to use plural here.

**Pull request merged**